### PR TITLE
LibJS: NonnullGCPtr all the things (actually just create() functions for now)

### PR DESCRIPTION
**Pull request by linusg**

### CODE REVIEW

File: Userland/Libraries/LibJS/Runtime/Array.h [hunk 1]
```diff
@@ -21,11 +21,11 @@ class Array : public Object {
     JS_OBJECT(Array, Object);
 
 public:
-    static ThrowCompletionOr<Array*> create(Realm&, u64 length, Object* prototype = nullptr);
-    static Array* create_from(Realm&, Vector<Value> const&);
+    static ThrowCompletionOr<NonnullGCPtr<Array>> create(Realm&, u64 length, Object* prototype = nullptr);
```
Comment (davidot) [inline Userland/Libraries/LibJS/Runtime/Array.h:24]: Technically this commit title should be `LibJS: Convert Array::create{,_from}() to NonnullGCPtr` right? (The comma is missing currently)

**Review by davidot — APPROVED** [review]
Mostly mechanical change so I'll leave the most up to the compiler, just one tiny thing about a commit title. But I'll approve on the condition of CI passing anyway.

I personally think it's a bit unfortunate we lose `auto*` clearly indicating we have a pointer and that there is a bunch of `.ptr()` calls now. But I assume those will disappear when everything is converted over?

File: Userland/Libraries/LibJS/Runtime/Array.h (hunk 1 above)
Comment (linusg) [reply Userland/Libraries/LibJS/Runtime/Array.h:24]: yep, fixed :+1:

**Comment by linusg** [comment]
> But I assume those will disappear when everything is converted over?

Right, that's the plan :^)

**Pull request merged**